DTS-STN/secure-client-hub
# Fix logout functionality

## [ADO-115882](https://dev.azure.com/VP-BD/DECD/_workitems/edit/115882)

### Description
Fixes [AB#115882](https://dev.azure.com/VP-BD/DECD/_workitems/edit/115882)

This PR adds logout functionality to the sign out button in the menu and the timeout modal on the `my-dashboard` page. The current global logout from ECAS doesn't redirect back upon logging out, so once logged out you'd have to open a new tab to log in again. This functionality is implemented by calling the `signOut` event from `next-auth` when the `Sign out` menu item is clicked, which ends the `next-auth` session, and then redirecting to `/auth/logout` (by default the `callbackURL` option from the `signOut` event only works with same origin URLs, hence the need to redirect to the logout page and redirect to an external link from there). In `/auth/logout`, the logout URL is obtained in `getServerSideProps` from `/lib/auth.js`, where I've added a new function to return a URL with the client ID and shared session ID. Upon loading the page, a user will see the loading spinner (for now) until they are redirected to the global log out page.

I've also changed the name and location of the `signedout` page to `/auth/login` to match the logout page (and because the naming of the page was confusing).

### What to test for/How to test
1. Pull in branch
2. Type `npm i`
3. Either go to the auth-dev branch which has been deployed with these changes **OR** fire up the proxy and continue with the below steps
4. Type `npm run dev`
5. Upon selecting a language, navigate through the login flow until you end back at `/my-dashboard`
6. Select `Sign out` from the menu
7. You should be redirected to `/auth/logout` and then to the global logout page
